### PR TITLE
Resolve site level redirects 

### PR DESCRIPTION
**Pull request by taranvohra**
This PR adds support for resolving site redirects configured on a site. The order is as follows
- First site content is always resolved to its canonical URL 
- Then we lookup the page using the `pathname`
  a.  If we find the page, good enough
  b.  If we don't find the page we check if there is a redirect at content (space) level 
  c. If we still don't find the page we check if there is a redirect at site level

Between (b) & (c), I suggest we do (b) first because content level redirects existed before.

### CODE REVIEW

File: packages/gitbook/src/app/(site)/fetch.ts [hunk 1]
```diff
@@ -88,20 +104,26 @@ async function resolvePage(
...
+
+        // If a page still can't be found, we try with the API, in case we have a redirect at site level.
+        const resolvedSiteRedirect = await getSiteRedirectBySource({
+            organizationId,
+            siteId,
+            source: rawPathname.startsWith('/') ? rawPathname : `/${rawPathname}`,
+            siteShareKey: input.shareKey,
+        });
+        if (resolvedSiteRedirect) {
+            return redirect(resolvedSiteRedirect.target);
```
Comment (taranvohra) [inline packages/gitbook/src/app/(site)/fetch.ts:125]: Using this [redirect](https://nextjs.org/docs/app/building-your-application/routing/redirecting#redirect-function) function returns a `307` which is fine for now but we'll need to think if we let users configure custom status codes for their redirects

File: packages/gitbook/src/app/(site)/fetch.ts [hunk 2]
```diff
@@ -88,20 +104,26 @@ async function resolvePage(
...
+        const resolved = await getRevisionPageByPath(spaceId, revisionId, rawPathname);
         if (resolved) {
             return resolvePageId(pages, resolved.id);
         }
+
+        // If a page still can't be found, we try with the API, in case we have a redirect at site level.
+        const resolvedSiteRedirect = await getSiteRedirectBySource({
+            organizationId,
+            siteId,
+            source: rawPathname.startsWith('/') ? rawPathname : `/${rawPathname}`,
```
Comment (SamyPesse) [inline packages/gitbook/src/app/(site)/fetch.ts:121]: The API should be improved to handle this and normalize input

File: packages/gitbook/src/app/(site)/fetch.ts (hunk 1 above)
Comment (SamyPesse) [reply packages/gitbook/src/app/(site)/fetch.ts:125]: We should keep everything as temporary redirect for now.

**Review by SamyPesse — APPROVED** [review]

File: packages/gitbook/src/app/(site)/fetch.ts (hunk 2 above)
Comment (taranvohra) [reply packages/gitbook/src/app/(site)/fetch.ts:121]: Sure 👍

File: packages/gitbook/src/app/(site)/fetch.ts (hunk 1 above)
Comment (gregberge) [reply packages/gitbook/src/app/(site)/fetch.ts:125]: It's not a 307, it's a redirect done client-side. Same issue as 404.  https://github.com/vercel/next.js/blob/5b7833e3da2b22e45e5b13c7001d4d728d093133/packages/next/src/client/components/redirect-boundary.tsx#L28-L37

**Pull request merged**